docroot/modules/custom/va_gov_backend/src/EventSubscriber/TableAuditViewsEventSubscriber.php

```diff
@@ -73,39 +71,82 @@ public function __construct(
    *
    * @throws \Exception
    */
-  public function preRender(ViewsPreRenderEvent $event): void {
+  public function preRender(ViewsPreRenderEvent $event) {
     $view = $event->getView();
     if ($view->id() === 'tables') {
-      foreach ($view->result as $key => $value) {
-        $node = va_gov_backend_get_parent_node($value->_entity);
-        if (empty($node)) {
-          $str = 'This is an orphan paragraph.';
-          $value->_entity->set('parent_field_name', $str);
-        }
-        else {
-          $link = Link::fromTextAndUrl($node->getTitle(), $node->toUrl())->toRenderable();
-          $content_type = $node->type->entity->label();
-          $va_gov_url = $this->vaGovUrl->getVaGovFrontEndUrlForEntity($node);
-          $va_gov_link = Link::fromTextAndUrl($va_gov_url, Url::fromUri($va_gov_url))->toRenderable();
-          $va_gov_link['#attributes'] = ['class' => 'va-gov-url'];
-          $section = $node->field_administration->entity;
-          $section_link = Link::fromTextAndUrl($section->label(), $section->toUrl())->toRenderable();
-          $value->_entity->set('parent_id', $this->renderer->renderRoot($link));
-          $value->_entity->set('parent_type', $content_type);
-          $value->_entity->set('parent_field_name', $this->renderer->renderRoot($va_gov_link));
-          $value->_entity->set('revision_id', $this->renderer->renderRoot($section_link));
+      foreach ($view->result as $value) {
+        /** @var \Drupal\paragraphs\Entity\Paragraph $paragraph */
+        $paragraph = $value->_entity;
+        $tree = [];
+        $top_parent = va_gov_backend_get_top_parent_entity($paragraph, $tree);
+        if ($top_parent) {
+          $is_in_current_node = $this->checkNodeRevisionUsesParagraph($top_parent, $tree);
+          if ($is_in_current_node) {
+            if ($top_parent->getEntityTypeId() === 'node') {
+              $node = $top_parent;
+              $link = Link::fromTextAndUrl($node->getTitle(), $node->toUrl())->toRenderable();
+              $content_type = $node->getType();
+              $va_gov_url = $this->vaGovUrl->getVaGovFrontEndUrlForEntity($node);
+              $va_gov_link = Link::fromTextAndUrl($va_gov_url, Url::fromUri($va_gov_url))->toRenderable();
+              $va_gov_link['#attributes'] = ['class' => 'va-gov-url'];
+              $section = $node->field_administration->entity;
+              $section_link = Link::fromTextAndUrl($section->label(), $section->toUrl())->toRenderable();
+              $value->_entity->set('parent_id', $this->renderer->renderRoot($link));
+              $value->_entity->set('parent_type', $content_type);
+              $value->_entity->set('parent_field_name', $this->renderer->renderRoot($va_gov_link));
+              $value->_entity->set('revision_id', $this->renderer->renderRoot($section_link));
+            }
+            if ($top_parent->getEntityTypeId() !== 'node') {
```

Comment:
shouldn't this be an else?  The first if checks that it is a node.

```suggestion
            else {
```

Reply:
This is a fallback for other entity types where the table para might be used in the future. Currently only used in nodes. I added a comment.

Reply:
Sorry, My point is that we have two blocks of code under separate if clauses,  One runs if it is a node, one runs if it is anything else.  That should be and if else, not two ifs.
![image](https://user-images.githubusercontent.com/5752113/209239138-193094ba-fec3-4dbf-86ae-7c6220cfb7a1.png)

If we aren't going to make it an else, then it should not be using string match and should be using a interface check.   Just making it an else would make it cleaner.

Reply:
This has not been addressed.

Reply:
I don't necessarily agree with this. However, it is done.